var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -781,36 +650,49 @@ def define_tpl_enable(cmake_var, spec_var=None):
                     define('Tpetra_INST_INT_LONG_LONG', gotype == 'long_long'),
                 ])
 
-        # disable due to compiler / config errors:
-        if spec.satisfies('%xl') or spec.satisfies('%xl_r'):
+        # ################# Kokkos ######################
+
+        if '+kokkos' in spec:
+            arch = Kokkos.spack_micro_arch_map.get(spec.target.name, None)
+            if arch:
+                options.append(define("Kokkos_ARCH_" + arch.upper(), True))
+
+            define_kok_enable = _make_definer("Kokkos_ENABLE_")
             options.extend([
-                define('Trilinos_ENABLE_Pamgen', False),
-                define('Trilinos_ENABLE_Stokhos', False),
+                define_kok_enable('CUDA'),
+                define_kok_enable('OPENMP' if spec.version >= Version('13')
+                                  else 'OpenMP'),
             ])
+            if '+cuda' in spec:
+                options.extend([
+                    define_kok_enable('CUDA_UVM', True),
+                    define_kok_enable('CUDA_LAMBDA', True),
+                    define_kok_enable('CUDA_RELOCATABLE_DEVICE_CODE', 'cuda_rdc')
+                ])
+                arch_map = Kokkos.spack_cuda_arch_map
+                options.extend(
+                    define("Kokkos_ARCH_" + arch_map[arch].upper(), True)
+                    for arch in spec.variants['cuda_arch'].value
+                )
 
-        if sys.platform == 'darwin':
-            options.append(define('Trilinos_ENABLE_FEI', False))
-            if '+stk' in spec:
-                cxx_flags.extend(['-DSTK_NO_BOOST_STACKTRACE'])
+        # ################# System-specific ######################
+
+        # Fortran lib (assumes clang is built with gfortran!)
+        if ('+fortran +mpi' in spec
+                and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
+            mpifc = Executable(spec['mpi'].mpifc)
+            libgfortran = mpifc('--print-file-name', 'libgfortran.a', output=str)
+            options.append(define(
+                'Trilinos_EXTRA_LINK_FLAGS',
+                '-L%s/ -lgfortran' % (libgfortran),
```

Comment:
This change is resulting in broken `Makefile.export.Trilinos`
```
balay@xsdk:/data/balay/spack/spack-stage/spack-stage-petsc-3.14.1-ucrioqpy3pna663lszxzcaj7wjs7jmsp/spack-src>cat simplemake 
include /data/balay/spack/opt/spack/linux-centos7-skylake_avx512/gcc-7.4.0/trilinos-13.0.1-fuialvl7b3ubw6f4tvuhzogqss6x62iy/lib/../include/Makefile.export.Trilinos
listlibs:
        -@echo ${Trilinos_LIBRARIES}balay@xsdk:/data/balay/spack/spack-stage/spack-stage-petsc-3.14.1-ucrioqpy3pna663lszxzcaj7wjs7jmsp/spack-src>make -f simplemake listlibs
/data/balay/spack/opt/spack/linux-centos7-skylake_avx512/gcc-7.4.0/trilinos-13.0.1-fuialvl7b3ubw6f4tvuhzogqss6x62iy/lib/../include/Makefile.export.Trilinos:33: *** missing separator.  Stop.
```
installed `Makefile.export.Trilinos` contains:
```
## Extra link flags (e.g., specification of fortran libraries)                                                                                                                                                     
Trilinos_EXTRA_LD_FLAGS=-L/nfs/apps/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-7.4.0-4kdemuwlzds2ofpkkz7yytgi7kyojuvz/lib/gcc/x86_64-pc-linux-gnu/7.4.0/../../../../lib64/libgfortran.a
/ -lgfortran;/usr/lib64/libm.so
```

[Makefile.export.Trilinos.txt](https://github.com/spack/spack/files/7140678/Makefile.export.Trilinos.txt)
[spack-build-out-header.txt](https://github.com/spack/spack/files/7140687/spack-build-out-header.txt)

Reply:
Fix at: https://github.com/spack/spack/pull/25882

Reply:
Ughhh looks like I'm missing `os.path.dirname(`?